h/test/presenters_test.py

```diff
@@ -399,9 +407,64 @@ def test_document_link_hostname_same_as_link_text(hostname_or_filename, href,
 
 
 @document_link_fixtures
+def test_document_link_hostname_same_as_link_text_ignores_when_href_missing(hostname_or_filename,
+                                                                            href,
+                                                                            title,
+                                                                            link_text):
+    """
+    It should keep the hostname the same when href is missing, even if it
+    is the same as the link text.
+    """
+    href.return_value = None
+    title.return_value = "www.example.com"
```

Comment:
It would be clearer to use a representative title, ie. a filename (eg. "example.pdf") rather than a domain name in this test, since 'hostname' is in fact a filename in this context.